tests/test_oauth2_provider.py

```diff
@@ -303,3 +304,36 @@ def test_token_endpoint_unauth(self):
         resp = self.provider.token_endpoint(request=areq.to_urlencoded())
         atr = TokenErrorResponse().deserialize(resp.message, "json")
         assert _eq(atr.keys(), ['error_description', 'error'])
+
+
+    def test_response_types(self):
+        authreq = AuthorizationRequest(state="state",
+                                       redirect_uri="http://example.com/authz",
+                                       client_id="client1",
+                                       response_type='id_token token')
+
+        self.provider.cdb = {
+            "client1": {
+                "client_secret": "hemlighet",
+                "redirect_uris": [("http://example.com/authz", None)],
+                'token_endpoint_auth_method': 'client_secret_post',
+                'response_types': ['token id_token', 'id_token']
+            }
+        }
+
+        res = self.provider.auth_init(authreq.to_urlencoded())
+        assert isinstance(res, dict) and "areq" in res
+
+        self.provider.cdb["client1"]['response_types'] = ['token id_token',
```

Comment:
This line just sets the response types to the same value it already has. It can be deleted.
If that is true lines 330 / 331 seem identical to 324/325 and can be deleted too.

Reply:
Changed the response_types to ['id_token token'] instead. So we check both variants.

Reply:
ok